### PR TITLE
Add service monitor support

### PR DESCRIPTION
**Pull request by arturdzm**
Adds the support for integration with Prometheus Operator through use of ServiceMonitor

Example CR:

```
apiVersion: appsody.dev/v1beta1
kind: AppsodyApplication
metadata:
  name: example-appsodyapplication
spec:
  ....
  monitoring:
    endpoints:
    - basicAuth:
        password:
          key: password
          name: metrics-liberty
        username:
          key: username
          name: metrics-liberty
      interval: 5s
      tlsConfig:
        insecureSkipVerify: true
    labels:
      k8s-app: ""
```

The only required field under monitoring is labels to match to correct Prometheus instance

### CODE REVIEW

**Comment by arturdzm** [comment]
Resolves #46

File: pkg/apis/appsody/v1beta1/zz_generated.openapi.go [hunk 1]
```diff
@@ -11,13 +11,13 @@ import (
...
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationAutoScaling": schema_pkg_apis_appsody_v1beta1_AppsodyApplicationAutoScaling(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationService":     schema_pkg_apis_appsody_v1beta1_AppsodyApplicationService(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationSpec":        schema_pkg_apis_appsody_v1beta1_AppsodyApplicationSpec(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationStatus":      schema_pkg_apis_appsody_v1beta1_AppsodyApplicationStatus(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationStorage":     schema_pkg_apis_appsody_v1beta1_AppsodyApplicationStorage(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.StatusCondition":               schema_pkg_apis_appsody_v1beta1_StatusCondition(ref),
+		"./pkg/apis/appsody/v1beta1.AppsodyApplication":            schema_pkg_apis_appsody_v1beta1_AppsodyApplication(ref),
```
Comment (navidsh) [inline pkg/apis/appsody/v1beta1/zz_generated.openapi.go]: Do you know why this one is different now?

File: pkg/controller/appsodyapplication/appsodyapplication_controller.go [hunk 2]
```diff
@@ -486,5 +494,30 @@ func (r *ReconcileAppsodyApplication) Reconcile(request reconcile.Request) (reco
...
+				return r.ManageError(err, appsodyv1beta1.StatusConditionTypeReconciled, instance)
+			}
+		} else {
+			sm := &prometheusv1.ServiceMonitor{ObjectMeta: defaultMeta}
+			err = r.DeleteResource(sm)
+			if err != nil {
+				reqLogger.Error(err, "Failed to delete ServiceMonitor")
+				return r.ManageError(err, appsodyv1beta1.StatusConditionTypeReconciled, instance)
+			}
+		}
```
Comment (navidsh) [inline pkg/controller/appsodyapplication/appsodyapplication_controller.go]: Can you please add code to log if `prometheus` is not supported here? `reqLogger.V(1).Info(...)`

File: pkg/apis/appsody/v1beta1/zz_generated.openapi.go (hunk 1 above)
Comment (arturdzm) [reply pkg/apis/appsody/v1beta1/zz_generated.openapi.go]: Probably something to do with sdk versions

**Comment by neeraj-laad** [comment]
@arturdzm @navidsh  I want to understand what value makes sense for the label specified here.

`k8s-app: ""`

**Comment by Kamran64** [comment]
@arturdzm 
Currently we only use the `APPSODY_PROJECT_NAME` placeholder which gets replaced with the users actual project name at deploy time. I believe this is also used as the application name (as the `selector` is pointing at the same name). As such, I've altered the PR's to implement this change to use `APPSODY_PROJECT_NAME` for the value of `k8s-app`.

Please could we get a review? See [Flask](https://github.com/appsody/stacks/pull/419), [Express](https://github.com/appsody/stacks/pull/403), [Functions](https://github.com/appsody/stacks/pull/402)

Thank you

**Comment by arturdzm** [comment]
Uisng `k8s-app` with `APPSODY_PROJECT_NAME ` should be fine,
the cluster prometheus  (openshift-monitoring) only listens to ServiceMonitors in that namespace.

AFAIK there is kabanero guide in the works for Prometheus install instructions  so  my point is that the label has to match the selector user will specify when they install their own Prometheus.
So users might have to add/change this label depending on their setup

**Pull request merged**